src/fuzzywuzzy_compatible/process.rs

```diff
@@ -0,0 +1,234 @@
+pub fn default_scorer(query: &&str, choice: String) -> Score {
+    wratio(query, &choice, true, true).try_into().unwrap()
+}
+
+/// The default processor used for functions in this module. Delegates to
+/// [full_process].
+///
+/// The `&&str` is a consequence of many functions generically accepting `A` and
+/// passing in `&A` to [Processor]s. and Rust disliking `str` without a
+/// reference.
+pub fn default_processor(s: &&str) -> String {
+    full_process(*s, false)
+}
+
+/// Quickly compare and return scored [Match]es of the choices against the
+/// query. Delegates to [extract_without_order_full].
+///
+/// ```
+/// # use core::convert::TryInto;
+/// # use fuzzywuzzy::primitives::{Match, Score};
+/// # use fuzzywuzzy::fuzzywuzzy_compatible::process::extract_without_order;
+/// let query = "bar";
+/// let choices = vec!["foo", "bar", "baz"];
+/// assert_eq!(extract_without_order(query, &choices), vec![Match{ item: "foo", score: 0.try_into().unwrap() }, Match{ item: "bar", score: 100.try_into().unwrap() }, Match{ item: "baz", score: 67.try_into().unwrap() }, ]);
+/// ```
+pub fn extract_without_order<'a>(query: &'a str, choices: &[&'a str]) -> Vec<Match<&'a str>> {
+    extract_without_order_full(
+        query,
+        choices,
+        default_processor,
+        default_scorer,
+        Score::new(0).unwrap(),
+    )
+    .into_iter()
+    .map(|s| Match {
+        item: *s.item,
+        score: s.score,
+    })
+    .collect()
+}
+
+// TODO: add to a doctest here....
+// let choices = vec![
+//     "new york mets vs chicago cubs",
+//     "chicago cubs vs chicago white sox",
+//     "philladelphia phillies vs atlanta braves",
+//     "braves vs mets",
+// ];
+// let expected_results = vec![
+//    ("new york mets vs chicago cubs".to_string(), 86u8),
+//    ("chicago cubs vs chicago white sox".to_string(), 86u8),
+//    ("philladelphia phillies vs atlanta braves".to_string(), 54u8),
+//    ("braves vs mets".to_string(), 57u8)
+// ];
+// assert_eq!(
+//     extract_without_order(
+//         "brave new cubs",
+//         choices,
+//         |s, b| s.into(), // an alternative to full_process.
+//         &wratio,
+//         0),
+//     expected_results);
+// ```
+
+/// Return scored [Match]es of the choices against the query with your choice of
+/// [Processor] and [Scorer].
+///
+/// ```
+/// # use core::convert::TryInto;
+/// # use fuzzywuzzy::primitives::{Match, Score};
+/// # use fuzzywuzzy::fuzzywuzzy_compatible::process::{ default_processor, default_scorer, extract_without_order_full };
+/// let query = "bar";
+/// let choices = vec!["foo", "bar", "baz"];
+/// assert_eq!(extract_without_order_full(query, &choices, default_processor, default_scorer, 0.try_into().unwrap()), vec![Match{ item: &"foo", score: 0.try_into().unwrap() }, Match{ item: &"bar", score: 100.try_into().unwrap() }, Match{ item: &"baz", score: 67.try_into().unwrap() }, ]);
+/// ```
+pub fn extract_without_order_full<'a, 'b, A: 'a, B, C, P, S>(
```

Comment:
Pedantic nitpicking: is `_full` a common suffix for functions providing more behavior controls in the signature?

Reply:
I don't know! It would be good to find some prior art on this before 1.0 but its probably fine for this PR. If you think we can do better, make an issue so we remember to check it out again.